### PR TITLE
[Kernel][Attention] Separate `Attention.kv_scale` into `k_scale` and `v_scale`

### PR DESCRIPTION
**Pull request by mgoin**
Since we already quantize `key_cache` and `value_cache` separately in PagedAttention, there is "free accuracy on the table" for FP8 KV Cache quantization as we could use separate per-tensor scales for each. 

The FlashInfer FP8 attention kernel also uses separate `k_scale` and `v_scale` values, so this PR is in preparation to enable that usage. Source: https://github.com/flashinfer-ai/flashinfer/blob/dc2c76f8577d8695112b61d1fd43ef88569272ef/python/flashinfer/decode.py#L98-L101

This PR will maintain backwards compatibility with FP8 model checkpoints that currently use `kv_scale` and duplicates that scale for both key+value if that is all that is available. However if a checkpoint has `k_scale` and `v_scale` present on the attention module, we will prefer those values.

### CODE REVIEW

File: vllm/model_executor/layers/quantization/fp8.py [hunk 1]
```diff
@@ -295,17 +297,37 @@ def process_weights_after_loading(self, layer: Module) -> None:
...
+                if not isinstance(kv_scale, float):
+                    raise ValueError("Only support per-tensor scaling factor "
+                                     "for fp8 KV cache")
+                layer._kv_scale = kv_scale
+                if (layer._kv_scale == 1.0
+                        and "e5m2" not in layer.kv_cache_dtype):
+                    print_warning_once(
+                        "Using KV cache scaling factor 1.0 for fp8_e4m3. This "
+                        "may cause accuracy issues. Please make sure kv-cache "
+                        "scaling factor is available in the fp8 checkpoint.")
```
Comment (comaniac) [inline vllm/model_executor/layers/quantization/fp8.py:327]: IIUC this is for back compatibility? If so I'd suggest make it explicitly, even with deprecate warning.

Another way I prefer more is only maintaining separated scales in model parameters. For example, we could remap kv_scale to key_scale, and and duplicate key_scale to value_scale here if we found that value_scale is 0, which is invalid anyways.

Comment (mgoin) [reply vllm/model_executor/layers/quantization/fp8.py:327]: Yes, I have light deprecation message elsewhere but will make it more explicit. I like the idea of just mapping kv_scale -> key_scale, will implement!

File: vllm/model_executor/models/llama.py [hunk 2]
```diff
@@ -460,18 +460,9 @@ def load_weights(self, weights: Iterable[Tuple[str, torch.Tensor]]):
...
-                            f"Found kv scale in the checkpoint (e.g. {name}), "
-                            "but not found the expected name in the model "
-                            f"(e.g. {remapped_kv_scale_name}). kv-scale is "
-                            "not loaded.")
-                        continue
-                    else:
-                        name = remapped_kv_scale_name
+                name = maybe_remap_kv_scale_name(name, params_dict)
+                if name is None:
+                    continue
```
[the remark this answers is not in the document]
Comment (mgoin) [reply vllm/model_executor/models/llama.py:465]: The function `maybe_remap_kv_scale_name` only returns None if the remapped name is not found in the params_dict, which is the same case that we skipped previously.

[the remark this answers is not in the document]
Comment (comaniac) [reply vllm/model_executor/models/llama.py:465]: Ah yeah you're right. Plz ignore this.

**Review by comaniac — APPROVED** [review]
LGTM

**Pull request merged**